packages/types/src/DidDetails.ts

```diff
@@ -139,75 +145,46 @@ export type DidServiceEndpoint = {
   /**
    * The identifier of the endpoint, without the leading did:kilt:<did_identifier> prefix.
    */
-  id: string
+  id: UriFragment
   /**
    * A list of service types the endpoint exposes.
    */
-  types: string[]
+  type: string[]
   /**
    * A list of URIs the endpoint exposes its services at.
    */
-  uris: string[]
+  serviceEndpoint: string[]
 }
 
 /**
- * An internal representation of data associated with a DID, equivalent to a DID document.
+ * A signature issued with a DID associated key, indicating which key was used to sign.
  */
-export interface IDidDetails {
-  /**
-   * The decentralized identifier (DID) to which the remaining info pertains.
-   */
-  uri: DidUri
+export type DidSignature = {
+  keyUri: DidResourceUri
+  signature: string
+}
 
-  /**
-   * Retrieves a particular public key record via its id.
-   *
-   * @param id The key ID, without the leading DID URI.
-   * @returns [[DidKey]] or undefined if no key with this id is present.
-   */
-  getKey(id: DidKey['id']): DidKey | undefined
+export interface DidDetails {
```

Comment:
This seems to overlap with `DidCreationDetails` quite a bit and in the bundle tests the light did details are even used for `getStoreTx` to generate the transaction for a full did.
Can / Should we try to bring them together?
Or is the distinction important, e.g. for creating a full did without lightdid details?

Reply:
While the interfaces are similar, they have important differences, and I don’t see a practical way to bring them together. The interfaces for creation define keys that do not need IDs, while the keys of the DidDetail have IDs. Moreover, the creation of light DIDs only supports two key types. 

An alternative approach would be to inline these type definitions. I don’t think this it’s a good API to give such a central position to a type of one parameter of one function.